### DISCUSSION TITLE
[Projects Beta] Existing iterations should have editable dates

### DISCUSSION THREAD

**Discussion by kevinwhoffman** [community/community#6776] [Projects and Issues] [answered]
## Overview

When the timeline of a release cycle changes, I want to edit the start/end dates of my existing iterations so that they reflect the new timeline.

## Current Outcome

Currently I have to delete the iterations and recreate them with the new start/end dates. For example, if I want to extend the existing iteration by one week, I have to delete it _and_ all of the iterations after it just to change the date.

Deleting an iteration also removes its connection to existing issues, so it becomes extra painful to go back and reconnect those issues to the new iterations.

<img width="1078" alt="Google Chrome-WP Migrate-20211026T184920" src="https://user-images.githubusercontent.com/1938671/138942286-19b9e9d8-e4c4-4d99-bf1d-4862f44b0a78.png">

## Desired Outcome

Ideally I would be able to click and edit the start/end dates of existing iterations without having to recreate them from scratch.

![Google Chrome-WP Migrate-20211026T185635](https://user-images.githubusercontent.com/1938671/138943410-9f5ef2dd-79cd-4700-b92c-202eca411fbc.png)

**Comment by joneff** [comment]
I think the main issue is that there is `<input type="date" />` inside a button and the click probably doesn't propagate. A hacky solution is to move the input out of the button and the remove a couple of styles to make it clickable and then one can create the the iterations even from past dates.

![image](https://user-images.githubusercontent.com/8048/140537353-5e4982db-063f-4af5-8bee-66e0e6b16ef0.png)

Note: this is not really a solution

**Reply by andreyrd** [reply]
You can't even change the date of the iteration to add, using the "Starts on" button, because of the same issue.

**Comment by azenMatt** [comment]
We are working on some improvements here to make this more flexible 👍 


How is having all iterations the same length working for you? Or do you also need those to be more flexible?

**Reply by kevinwhoffman** [reply]
I'd prefer the ability to have different lengths with custom start and end dates.

### Existing limitations of iteration field

For example, if we have a 2-week iteration, I want the iteration to start on a Monday, Nov 29 and end on the following Friday, Dec 10. The existing behavior of starting on a Monday and ending on a Sunday is awkward, and it does not reflect the true deadline of the iteration in terms of business days.

### Using single select field to overcome limitations of the iteration field

For now, the existing iteration field type is too limited and rigid for our needs. We've reverted to using a single select field to represent iterations in our current projects. Here's what that looks like:

<img width="1245" alt="Google Chrome-WP Migrate-20211125T172443" src="https://user-images.githubusercontent.com/1938671/143482522-27bbda4e-b08d-47ec-a13e-11defeff7a8b.png">

This makes it easier for me to:

- Communicate exact start/end dates of an iteration
- Adjust dates more easily without having to delete and recreate iterations
- Set start dates in the past, which is essential for anyone who begins using the new Projects beta in the middle of an existing cycle

In summary, I'd like to see the iteration field type behave exactly like a single select field with the addition of a date picker to set the date range of each iteration.

**Reply by azenMatt** [reply]
Thanks for the thoughts @kevinwhoffman - this is the way we are leaning as well. 

cc @six7

**Reply by abernix** [reply]
My teams would also love if "month" could be added as an iteration interval.  Without it, my teams still elect to use the single-select box with manually curated values. 

(The above suggestions sound like great improvements, too!)

**Comment by azenMatt** [comment, marked as answer]
Hi everyone - we just shipped some updates that make it easier to work with iteration dates! 

https://github.blog/changelog/2022-02-02-the-new-github-issues-02-02-update/

Hopefully this helps!
